### PR TITLE
[lldb] Fix incorrect logical operator in 'if' condition check (NFC)

### LINKED ISSUE #89195: lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:89: possible && and || mixup ?
Source code analyser cppcheck says:

> trunk/lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:89:11: warning: Identical inner 'if' condition is always true. [identicalInnerCondition]
> trunk/lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:91:16: warning: Identical inner 'if' condition is always true. [identicalInnerCondition]

Source code is

    if (!has_class_name && !has_interpreter_dict && !script_obj) {
      if (!has_class_name)
        return create_error("Missing script class name.");
      else if (!has_interpreter_dict)
        return create_error("Invalid script interpreter dictionary.");
      else
        return create_error("Missing scripting object.");

I think the original coder wanted something like:

    if (!has_class_name || !has_interpreter_dict || !script_obj) {
      if (!has_class_name)
        return create_error("Missing script class name.");
      else if (!has_interpreter_dict)
        return create_error("Invalid script interpreter dictionary.");
      else
        return create_error("Missing scripting object.");

### PR DESCRIPTION
**Pull request by xgupta**
The condition checking for missing class name, interpreter dictionary, and script object incorrectly used logical AND (&&), which could never be true to enter the 'if' block. 

This commit uses separate if conditions for each class name, interpreter dictionary, and script object.

Cought by cppcheck -
lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:89:11: warning: Identical inner 'if' condition is always true. [identicalInnerCondition]
lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:91:16: warning: Identical inner 'if' condition is always true. [identicalInnerCondition]

Fix #89195

### CODE REVIEW

**Review by JDevlieghere — CHANGES_REQUESTED** [review]
Similar comment as https://github.com/llvm/llvm-project/pull/94775#issuecomment-2155285000

**Comment by xgupta** [comment]
Gentile ping!

File: lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h [hunk 1]
```diff
@@ -85,13 +85,17 @@ class ScriptedPythonInterface : virtual public ScriptedInterface {
...
-      if (!has_class_name)
-        return create_error("Missing script class name.");
-      else if (!has_interpreter_dict)
-        return create_error("Invalid script interpreter dictionary.");
-      else
-        return create_error("Missing scripting object.");
+
+    if (!has_class_name) {
+      return create_error("Missing script class name.");
+    }
```
Comment (DavidSpickett) [inline lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h:91]: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

Comment (xgupta) [reply lldb/source/Plugins/ScriptInterpreter/Python/Interfaces/ScriptedPythonInterface.h]: Sure, thanks.

**Comment by xgupta** [comment]
ping for review comment.

**Review by bulbazord — APPROVED** [review]
The change looks good to me. Please make sure you update the commit message so it accurately reflects the change since there were a few revisions. :)

**Comment by DavidSpickett** [comment]
The code looks ok but as landed it turned `A && B && C` into `A || B || C`, despite that looking logical.

@medismailben can you figure out the intent here, I'm assuming this was your code originally.

**Pull request merged**